src/Pos/Explorer/Web/Server.hs

```diff
@@ -349,26 +349,34 @@ getAddressSummary
 getAddressSummary cAddr = do
     addr <- cAddrToAddr cAddr
 
-    when (isAddressUnknown addr) $
+    when (getAddressType addr == CUnknownAddress) $
         throwM $ Internal "Unknown address type"
 
-    balance <- mkCCoin . fromMaybe (mkCoin 0) <$> EX.getAddrBalance addr
+    balance <- fromMaybe (mkCoin 0) <$> EX.getAddrBalance addr
     txIds <- getNewestFirst <$> EX.getAddrHistory addr
     transactions <- forM txIds $ \id -> do
         extra <- getTxExtraOrFail id
         tx <- getTxMain id extra
         pure $ makeTxBrief tx extra
-    pure CAddressSummary {
-        caAddress = cAddr,
-        caType = getAddressType addr,
-        caTxNum = fromIntegral $ length transactions,
-        caBalance = balance,
-        caTxList = transactions
-    }
+
+    isRedeemed <- do
+        if getAddressType addr /= CRedeemAddress then pure Nothing else do
+            redeemAddressCoinPairs <- getRedeemAddressCoinPairs
+            let mGenesisBalance = snd <$> (head $
```

Comment:
Sorry, my bad, the `head` function is good, using `maybeThrow` would make it more readable?